panimg/image_builders/nrrd.py

```diff
@@ -0,0 +1,118 @@
+import re
+from collections import defaultdict
+from pathlib import Path
+from typing import DefaultDict, Iterator, List, Set
+
+import SimpleITK
+
+from panimg.exceptions import UnconsumedFilesException
+from panimg.models import SimpleITKImage
+
+MAGIC_REGEX = re.compile("NRRD([0-9]{4})")
+DATAFILE_REGEX = re.compile(r"data\s?file:", flags=re.IGNORECASE)
+
+
+class InvalidNrrdFileError(Exception):
+    def __init__(self, message: str = "Not a NRRD image file"):
+        super().__init__(message)
+
+
+def verify_single_file_nrrd(file: Path) -> bool:
+    """
+    Reads a file and raises an InvalidNrrdFileError if that file is not a NRRD file.
+    Returns True if the file is a valid NRRD file without external data file (data
+    embedded in the same file). Returns False if the file is valid NRRD file but
+    has one or more external data files (detached header mode).
+    """
+    with file.open("rb") as fp:
+        # Read first 8 characters to check whether this is a NRRD file
+        try:
+            preamble = fp.read(8).decode("ASCII")
+        except UnicodeDecodeError:
+            raise InvalidNrrdFileError
+
+        match = MAGIC_REGEX.fullmatch(preamble)
+        if match is None:
+            raise InvalidNrrdFileError
+
+        # Check whether we support this version of the NRRD format
+        nrrd_version = int(match.group(1))
+        if nrrd_version < 1 or nrrd_version > 5:
+            raise InvalidNrrdFileError("File format not supported")
+
+        # Discard rest of preamble line
+        fp.readline(1000)
```

Comment:
Is the choice of 1000 bytes and lines arbitrary or is there something in the file spec on this?

Reply:
That's an arbitrary value, I followed the implementation of header parsing function in metaio_utils.py in this (which actually uses 10,000, will adjust it to that)